tests/testthat/test-seq_linter.R

```diff
@@ -123,6 +123,24 @@ test_that("reverse seq is ok", {
   )
 })
 
+test_that("finds potential sequence() replacements", {
+  linter <- seq_linter()
+  lint_msg <- rex::rex("Use sequence()")
+
+  expect_lint(
+    "unlist(lapply(x, seq_len))",
```

Comment:
The lint now only covers `unlist(lapply(x, seq_len))`.

What about

 - `unlist(lapply(x, seq))`
 - `unlist(lapply(x, seq, by = 2))`, and other variants equivalent to using the `from=` and/or `by=` arguments of `sequence()`?
 - I _don't_ think it's worth trying to find examples like `lapply(x, \(xi) seq(xi))` which would be covered by `unnecessary_lambda_linter()`. But are there any examples that _do_ require looking into a lambda to replicate a `sequence()`-able call?

(it would be nice to get a sense of how many hits these are getting before investing too much time -- and also feel free to earmark some of this as a TODO in a follow-up issue, though `lapply(x,seq)` is trivial enough to be done here)

Reply:
> `unlist(lapply(x, seq, by = 2))`, and other variants equivalent to using the `from=` and/or `by=` arguments of sequence()?

I don't think there is a direct equivalence:

- `from` has a difference behaviour in `seq()` and `sequence()`

  ``` r
  unlist(lapply(1:5, seq, from = 2))
  #>  [1] 2 1 2 2 3 2 3 4 2 3 4 5

  sequence(1:5, from = 2)
  #>  [1] 2 2 3 2 3 4 2 3 4 5 2 3 4 5 6
  ```

  <sup>Created on 2024-06-22 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

- As far as I can tell, `by=` cannot be used on its own without `from=`, which brings us back to the previous case

  ``` r
  seq(20, by = 2)
  #> Error in seq.default(20, by = 2): wrong sign in 'by' argument
  ```

  <sup>Created on 2024-06-22 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

I have edited the xpath to make sure this type of call doesn't lint.